### PR TITLE
Footer Overflow fixed

### LINKED ISSUE #37: Bug of Footer on Mobile Devices
- [x] Some unnecessary stateful widgets.
- [x] Responsive bug on mobile devices. 

You can check the mobile device's view down below. (iPhone X)

Screen Shot 2021-02-07 at 16 55 58

### PR DESCRIPTION
**Pull request by Adem68**
Footer overflow fixed

Fixes #37

Full Size | Mobile
------------ | -------------
image | image

Before | After
------------ | -------------
image | image

### CODE REVIEW

File: lib/pages/home_page.dart [hunk 1]
```diff
@@ -92,7 +92,7 @@ class _HomePageState extends State<HomePage> {
                 ),
                 SectionTitle(title: 'Konuşmacılar'),
                 SpeakersSection(),
-                FooterView(),
+                FooterSection(),
```
Comment (bgoktugozdemir) [inline lib/pages/home_page.dart:95]: Try to make it const.

File: lib/widgets/footer_section.dart [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+
+class FooterSection extends StatelessWidget {
+  const FooterSection({Key key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return ResponsiveBuilder(
+      largeWidget: LargeWidget(),
+      mediumWidget: MediumWidget(),
+      smallWidget: SmallWidget(),
```
Comment (bgoktugozdemir) [inline lib/widgets/footer_section.dart:16]: Try to make these three widgets const.

File: lib/widgets/footer_section.dart [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+            children: [
+              Container(
+                decoration: BoxDecoration(
+                  image: DecorationImage(
+                    image: AssetImage(Assets.footerBackground),
+                    fit: BoxFit.cover,
+                  ),
+                ),
+              ), // background
+              Container(
```
Comment (bgoktugozdemir) [inline lib/widgets/footer_section.dart:42]: Why did we add this container? It doesn't have any height or width. Is it working?

File: lib/widgets/footer_section.dart [hunk 4]
```diff
@@ -0,0 +1,421 @@
...
+                color: Colors.black.withOpacity(0.75),
+              ), // black filter
+              Center(
+                child: Column(
+                  crossAxisAlignment: CrossAxisAlignment.center,
+                  children: [
+                    Padding(
```
Comment (bgoktugozdemir) [inline lib/widgets/footer_section.dart:49]: Try to make it const.

File: lib/widgets/footer_section.dart [hunk 5]
```diff
@@ -0,0 +1,421 @@
...
+                    ),
+                    Padding(
+                      padding: const EdgeInsets.only(top: 30, bottom: 10),
+                      child: InkWell(
+                        onTap: () =>
+                            launch('mailto:hackathon@flutterturkiye.org'),
+                        child: Row(
+                          mainAxisAlignment: MainAxisAlignment.center,
+                          children: [
+                            Icon(
```
Comment (bgoktugozdemir) [inline lib/widgets/footer_section.dart:65]: Try to make it const.

File: lib/widgets/footer_section.dart [hunk 6]
```diff
@@ -0,0 +1,421 @@
...
+                      ),
+                      child: Image.asset(
+                        Assets.logo,
+                      ),
+                    ),
+                    Padding(
+                      padding: const EdgeInsets.only(top: 30, bottom: 10),
+                      child: InkWell(
+                        onTap: () =>
+                            launch('mailto:hackathon@flutterturkiye.org'),
```
Comment (bgoktugozdemir) [inline lib/widgets/footer_section.dart:61]: Let's use **canLaunch** method first.

File: lib/widgets/footer_section.dart [hunk 7]
```diff
@@ -0,0 +1,421 @@
...
+                        child: Row(
+                          mainAxisAlignment: MainAxisAlignment.center,
+                          children: [
+                            Icon(
+                              Icons.mail,
+                              color: Colors.white,
+                            ),
+                            Padding(
```
Comment (bgoktugozdemir) [inline lib/widgets/footer_section.dart:69]: Let's try to make Padding const. If it won't work, Text can be const.

File: lib/widgets/footer_section.dart [hunk 8]
```diff
@@ -0,0 +1,421 @@
...
+                                  fontSize: 18,
+                                  fontFamily: 'Montserrat',
+                                ),
+                              ),
+                            ),
+                          ],
+                        ),
+                      ),
+                    ),
+                    Divider(
```
Comment (bgoktugozdemir) [inline lib/widgets/footer_section.dart:85]: Try to make it const.

File: lib/widgets/footer_section.dart [hunk 9]
```diff
@@ -0,0 +1,421 @@
...
+                      indent: 100,
+                      endIndent: 100,
+                      color: Colors.white,
+                      thickness: 0.5,
+                    ),
+                    Align(
```
Comment (bgoktugozdemir) [inline lib/widgets/footer_section.dart:91]: Try to make it const.

File: lib/widgets/footer_section.dart [hunk 10]
```diff
@@ -0,0 +1,421 @@
...
+                        child: Text(
+                          '© 2021 Flutter Türkiye',
+                          style: const TextStyle(
+                            color: Colors.white,
+                            fontSize: 16,
+                          ),
+                        ),
+                      ),
+                    ),
+                    BottomView(),
```
Comment (bgoktugozdemir) [inline lib/widgets/footer_section.dart:107]: Try to make it const.

File: lib/widgets/footer_section.dart [hunk 11]
```diff
@@ -0,0 +1,421 @@
...
+            children: [
+              Container(
+                decoration: BoxDecoration(
+                  image: DecorationImage(
+                    image: AssetImage(Assets.footerBackground),
+                    fit: BoxFit.cover,
+                  ),
+                ),
+              ), // background
+              Container(
```
Comment (bgoktugozdemir) [inline lib/widgets/footer_section.dart:138]: Why did we add this container? It doesn't have any height or width. Is it working?

**Review by bgoktugozdemir — CHANGES_REQUESTED** [review]
I guess all these three widgets are almost the same except for the values. Let's create widgets for common parts of these widgets.

File: lib/widgets/footer_section.dart (hunk 3 above)
Comment (Adem68) [reply lib/widgets/footer_section.dart:42]: This container added for black filter. It has same height with top widget.

File: lib/widgets/footer_section.dart (hunk 11 above)
Comment (Adem68) [reply lib/widgets/footer_section.dart:138]: This container added for black filter. It has same height with top widget.

File: lib/pages/home_page.dart [hunk 12]
```diff
@@ -5,7 +5,7 @@ import '../core/components/app_bar/app_bar_action_button.dart';
 import '../core/components/app_bar/base_app_bar.dart';
 import '../core/components/drawer/base_drawer.dart';
 import '../core/model/app_bar_and_drawer_item_model.dart';
-import '../widgets/footer_view.dart';
+import '../widgets/footer_section.dart';
```
Comment (salihgueler) [inline lib/pages/home_page.dart:8]: Let's merge this and I will fix all the paths.

File: lib/widgets/footer_section.dart [hunk 13]
```diff
@@ -0,0 +1,417 @@
...
```
Comment (salihgueler) [inline lib/widgets/footer_section.dart:42]: Why didn't you use the decoration?

File: lib/widgets/footer_section.dart [hunk 14]
```diff
@@ -0,0 +1,422 @@
...
+    );
+  }
+}
+
+class SmallWidget extends StatelessWidget {
+  const SmallWidget({Key key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Align(
```
Comment (bgoktugozdemir) [inline lib/widgets/footer_section.dart:26]: Container already has an alignment. Let's remove this and add it into Container.

File: lib/widgets/footer_section.dart [hunk 15]
```diff
@@ -0,0 +1,422 @@
...
+                    Padding(
+                      padding: const EdgeInsets.only(
+                        top: 40.0,
+                      ),
+                      child: Image.asset(
+                        Assets.logo,
+                      ),
+                    ),
+                    Padding(
+                      padding: const EdgeInsets.only(top: 30, bottom: 10),
```
Comment (bgoktugozdemir) [inline lib/widgets/footer_section.dart:60]: Let's update these values.

File: lib/widgets/footer_section.dart [hunk 16]
```diff
@@ -0,0 +1,422 @@
...
+                      child: InkWell(
+                        onTap: () =>
+                            launch('mailto:festival@flutterturkiye.org'),
```
Comment (bgoktugozdemir) [inline lib/widgets/footer_section.dart:63]: Let's use canLaunch method first.

**Review by bgoktugozdemir — CHANGES_REQUESTED** [review]

**Review by bgoktugozdemir — APPROVED** [review]

**Review by salihgueler — APPROVED** [review]

**Pull request merged**